### ISSUE TITLE
na_ontap_quotas fails when using loop with multiple qtrees

### ISSUE THREAD

**Issue by thanos-k** [ansible-collections/netapp.ontap#44]
### Summary

When setting quotas on multiple qtrees with a loop occasionally (not always) the task fails when there is a change in quota. 
The taks does not fail on the item that is changed but on the next 1-2 items because `resize` that is triggered by the `activate_quota_on_change` parameter has not completed.  
Not certain if the size of the quota or the size/number of files in the qtree play a role. 

### Component Name

yum

### Ansible Version

```console
$ ansible --version

ansible 2.9.27
  config file = /home/thanosk/it-ansible/ansible.cfg
  configured module search path = ['/home/thanosk/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/thanosk/it-ansible/ansible.venv/lib64/python3.6/site-packages/ansible
  executable location = /home/thanosk/it-ansible/ansible.venv/bin/ansible
  python version = 3.6.8 (default, Nov 16 2020, 16:55:22) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

### ONTAP Collection Version

```console
"collection_info": {
  "namespace": "netapp",
  "name": "ontap",
  "version": "21.15.1",
....
}
```

### ONTAP Version

```console
::> version
NetApp Release 9.9.1P4: Mon Oct 11 09:20:31 UTC 2021
```

### Playbook

```console
task: 

    - name: Update Quota
      netapp.ontap.na_ontap_quotas:
        vserver: "{{ vserver }}"
        policy: "{{ policy }}"
        volume: "{{ volume }}"
        type: tree
        quota_target: "/vol/{{ volume }}/{{ item.qtree_name }}"
        disk_limit: "{{ item.qtree_disk_limit*1048576 }}"
        set_quota_status: True
        activate_quota_on_change: resize
        state: "{{ item.qtree_state | default('present') }}"
        hostname: "{{ ontap_hostname }}"
        username: "{{ ontap_username }}"
        password: "{{ ontap_pass }}"
        https: yes
        validate_certs: no
        use_rest: always
      delegate_to: localhost
      loop: "{{ qtree_shares }}"
      tags: quota
```

### Steps to Reproduce

Run the above task with the following vars making certain that the first qtree has the quota changed (from 50 to 51 in this case)

```yaml
qtree_shares:
  - { qtree_name: IT              , qtree_disk_limit: 51    }
  - { qtree_name: ad              , qtree_disk_limit: 50    }
```

### Expected Results

The first qtree should have the quota changed and the second one return as no change 

### Actual Results

The first qtree quota changed correctly. 
```console
changed: [******** -> localhost] => (item={'qtree_name': 'IT', 'qtree_disk_limit': 51}) => { ....
```
but for the second qtree even though the quota was not supposed to change as it was already correctly set
```console
failed: [******** -> localhost] (item={'qtree_name': 'ad', 'qtree_disk_limit': 50}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "activate_quota_on_change": "resize",
            "cert_filepath": null,
            "disk_limit": "52428800",
            "feature_flags": {},
            "file_limit": null,
            "hostname": "********",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "perform_user_mapping": null,
            "policy": "Group",
            "qtree": "",
            "quota_target": "/vol/Group/ad",
            "set_quota_status": true,
            "soft_disk_limit": null,
            "soft_file_limit": null,
            "state": "present",
            "threshold": null,
            "type": "tree",
            "use_rest": "always",
            "username": "admin",
            "validate_certs": false,
            "volume": "Group",
            "vserver": "****"
        }
    },
    "item": {
        "qtree_disk_limit": 50,
        "qtree_name": "ad"
    },
    "msg": "Error setting quota-on for Group: NetApp API failed. Reason - 13001:Failed to load job for \"quota on\" performed for quota policy \"Group\" on volume \"Group\" in Vserver \"****\": Another quota operation is currently in progress for volume \"Group\" in Vserver \"****\". Use the \"quota show\" command to view the progress of the job. When it is complete, try the command again. "
```
```

**Comment by lonico** [comment]
Yes, it's an issue in the code.  We're interpreting any status different from 'on' as a need to force the status back to on.
We'll address it.   Though we don't know how long the resize (or reinitialize) operation will take so this may serialize the execution.

Maybe it would make sense to loop on setting the quotas, but have a separate task to trigger the resize.

**Comment by thanos-k** [comment]
@lonico thanks for the answer. The same problem exists obviously when both quotas are changed,
so serializing the execution and waiting for the resize to end might be the way to go.   

I can create a custom handler to execute the resize.  Is there a way to do it with the `na_ontap_quotas` module or should I use a simple `na_ontap_command` . Either way it is fine but am just curious.

**Comment by carchi8py** [comment]
We have an Internal Story DEVOPS-4858 open for this issue.

**Comment by wenjun666** [comment]
Hi @thanos-k, I can't reproduce the error, here are my tasks:

changed: [localhost] => (item={'qtree_name': 'q1', 'qtree_disk_limit': 52}) => {
    "ansible_loop_var": "item",
    "changed": true,
    "invocation": {
        "module_args": {
            "activate_quota_on_change": "resize",
            "cert_filepath": null,
            "disk_limit": "54525952",
            "feature_flags": {},
            "file_limit": null,
            "hostname": "***",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "perform_user_mapping": null,
            "policy": "test_quota_policy",
            "qtree": "",
            "quota_target": "/vol/vol1/q1",
            "set_quota_status": true,
            "soft_disk_limit": null,
            "soft_file_limit": null,
            "state": "present",
            "threshold": null,
            "type": "tree",
            "use_rest": "always",
            "username": "***",
            "validate_certs": false,
            "volume": "vol1",
            "vserver": "***"
        }
    },
    "item": {
        "qtree_disk_limit": 52,
        "qtree_name": "q1"
    },
    "warnings": [
        "Using ZAPI for netapp.ontap.na_ontap_quotas, ignoring 'use_rest: always'."
    ]
}

changed: [localhost] => (item={'qtree_name': 'q2', 'qtree_disk_limit': 53}) => {
    "ansible_loop_var": "item",
    "changed": true,
    "invocation": {
        "module_args": {
            "activate_quota_on_change": "resize",
            "cert_filepath": null,
            "disk_limit": "55574528",
            "feature_flags": {},
            "file_limit": null,
            "hostname": "***",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "perform_user_mapping": null,
            "policy": "test_quota_policy",
            "qtree": "",
            "quota_target": "/vol/vol1/q2",
            "set_quota_status": true,
            "soft_disk_limit": null,
            "soft_file_limit": null,
            "state": "present",
            "threshold": null,
            "type": "tree",
            "use_rest": "always",
            "username": "***",
            "validate_certs": false,
            "volume": "vol1",
            "vserver": "***"
        }
    },
    "item": {
        "qtree_disk_limit": 53,
        "qtree_name": "q2"
    },
    "warnings": [
        "Using ZAPI for netapp.ontap.na_ontap_quotas, ignoring 'use_rest: always'."
    ]
}

**Comment by carchi8py** [comment]
This was fixed with 21.20.0
